src/app/_api/fetchPodcastEpisodes.ts

```diff
@@ -0,0 +1,35 @@
+import { RequestCookie } from "next/dist/compiled/@edge-runtime/cookies";
+import { GRAPHQL_API_URL } from "@/app/_api/shared";
+import { PODCAST_EPISODES } from "@/app/_graphql/podcastepisodes";
+import { Config } from "@/payload/payload-types";
+
+export async function fetchPodcastEpisodes(collection: keyof Config["collections"]) {
+
+  if (!collection) throw new Error("Collection not found");
+
+  let token: RequestCookie | undefined;
+
+  // if (draft) {
+  //   const { cookies } = await import("next/headers")
+  //   token = cookies().get(payloadToken)
+  // }
+
+  const doc = await fetch(`${GRAPHQL_API_URL}/api/graphql`, {
+    method: "POST",
```

Comment:
If you're going to do this often, it might be better to create a helper function that receives the graphql query and deals with the boilerplate of setting method, headers, etc. Then you can just use it like `await request(PODCAST_EPISODES)`

Reply:
great idea, will encapsulate this